GamepadEventDrivenInputAPI/explainer.md

```diff
@@ -0,0 +1,237 @@
+
+    for (const axisIndex of changeEvent.axesChanged) {
+      const snapshotAxisValue = snapshot.axes[axisIndex];
+      const liveAxisValue = liveGamepad.axes[axisIndex];
+      console.log(`Axis ${axisIndex} on gamepad ${snapshot.index} changed to ${snapshotAxisValue} (live: ${liveAxisValue})`);
+    }
+
+    // Analog button changes (ex: triggers).
+    for (const buttonIndex of changeEvent.buttonsValueChanged) {
+      const snapshotButtonValueChanged = snapshot.buttons[buttonIndex].value;
+      const liveButtonsValueChanged = liveGamepad.buttons[buttonIndex].value;
+      console.log(`button ${buttonIndex} on gamepad ${snapshot.index} changed to value ${snapshotButtonValueChanged} (live: ${liveButtonValueChanged})`);
+    }
+
+    // Binary buttons that were pressed.
+    for (const buttonIndex of changeEvent.buttonsPressed) {
+      const snapshotButtonPressedValue = snapshot.buttons[buttonIndex].pressed;
+      const liveButtonPressedValue = liveGamepad.buttons[buttonIndex].pressed;
+      console.log(`button ${buttonIndex} on gamepad ${snapshot.index} changed to value ${snapshotButtonPressedValue} (live: ${liveButtonPressedValue}`);
+    }
+
+    // Binary buttons that were released.
+    for (const buttonIndex of changeEvent.buttonsReleased) {
+      const snapshotButtonReleasedValue = snapshot.buttons[buttonIndex].released;
+      const liveButtonReleasedValue = liveGamepad.buttons[buttonIndex].released;
+      console.log(`button ${buttonIndex} on gamepad ${snapshot.index} changed to value ${snapshotButtonReleasedValue} (live: ${liveButtonReleasedValue}`);
+    }
+  };
+};
+```
+
+## Alternatives considered
+`gamepadinputchange` event: Similar to `rawgamepadinputchange` event but instead the `getCoalescedEvents()` method is used to return a sequence of events that have been coalesced (combined) together.  While `gamepadinputchange` reduces the number of events by coalescing them, this approach introduces latency and may result in missed intermediate states, making it unsuitable for scenarios requiring immediate responsiveness. This event was proposed in the [Original Proposal](https://docs.google.com/document/d/1rnQ1gU0iwPXbO7OvKS6KO9gyfpSdSQvKhK9_OkzUuKE/edit?pli=1&tab=t.0).
+
+## Accessibility, Privacy, and Security Considerations
+To prevent abuse and fingerprinting, a ["gamepad user gesture"](https://www.w3.org/TR/gamepad/#dfn-gamepad-user-gesture) will be required before `RawGamepadInputChange` events start firing (e.g., pressing a button).
+
+Limit Persistent Tracking (fingerprinting): `rawgamepadinputchange` event will not expose any new state that is not already exposed by polling [Fingerprinting in Web](https://www.w3.org/TR/fingerprinting-guidance/).
+
+## Stakeholder Feedback / Opposition
+Firefox: No Signal
+
+Safari: No Signal
+
+Web Developers: Positive
+[Should fire events instead of using passive model](https://github.com/w3c/gamepad/issues/4),
+
+[Gamepad Other Events](https://w3c.github.io/gamepad/#other-events)
```

Comment:
Let's not link to the standard to signal developer interest.  The standard is meant for browser implementors -- not necessarily web developers.  Delete https://w3c.github.io/gamepad/#other-events, but we can keep the link to the issue which has reports from developers

Reply:
Here are some other links where developers want events:

Stack Overflow:
https://stackoverflow.com/questions/70788613/how-to-create-a-generic-joystick-gamepad-event-in-javascript
https://stackoverflow.com/questions/72294832/using-gamepad-api-via-events-rather-than-polling

Some libraries:
https://github.com/alvaromontoro/gamecontroller.js
https://github.com/blovato/Gamepad-Controller